### DISCUSSION TITLE
set dag status to failed but run all tasks

### DISCUSSION THREAD

**Discussion by waardd** [apache/airflow#29716] [Q&A]
I have a dag with several tasks.
The last task is to send me a message with the outcome.
I set a rule that says: trigger_rule="all_done" so that the step always gets done even when a task fails or is skipped.
But now that i have connected to prometheus and see metrics, i can not see if a task dag is failed because of a failure or skipped step because it is always success.

How can i still run the last task (even if others fail or skip) but still get the dag to state failed so i can monitor this?

**Comment by potiuk** [comment]
For now, you can run your last task and fail it conditional by checking status of other tasks (for example via REST API). Or I believe you can have another (failing) task watching all others with "one_failed" and fail it (I believe it should work).

In the (hopefully near - 2.6 or 2.7) future you will be able to use AIP-52 for that case https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-52+Automatic+setup+and+teardown+tasks
